packages/ocap-kernel/src/remotes/OcapURLManager.test.ts

```diff
@@ -0,0 +1,310 @@
+        [],
+      );
+      expect(mockRemoteHandle.redeemOcapURL).toHaveBeenCalledWith(url);
+    });
+
+    it('should pass parsed hints to remoteFor for remote OCAP URL', async () => {
+      const url = 'ocap:def456@remote-peer-id,relay1,relay2';
+      const kref = await ocapURLManager.redeemOcapURL(url);
+
+      expect(kref).toBe('ko456');
+      expect(mockRemoteManager.remoteFor).toHaveBeenCalledWith(
+        'remote-peer-id',
+        ['relay1', 'relay2'],
+      );
+      expect(mockRemoteHandle.redeemOcapURL).toHaveBeenCalledWith(url);
+    });
+
+    it('should throw for invalid OCAP URL', async () => {
+      await expect(ocapURLManager.redeemOcapURL('invalid-url')).rejects.toThrow(
+        'unparseable URL',
+      );
+    });
+
+    it('should throw if remote comms is not initialized', async () => {
+      (mockRemoteManager.getRemoteComms as Mock).mockImplementation(() => {
+        throw new Error('Remote comms not initialized');
+      });
+
+      await expect(
+        ocapURLManager.redeemOcapURL('ocap:abc123@local-peer-id'),
+      ).rejects.toThrow('Remote comms not initialized');
+    });
+  });
+
+  describe('issuer service', () => {
+    it('should issue URL through issuer service', async () => {
+      // Since we're testing integration with krefOf which requires a special
+      // object structure, we'll test the underlying mechanism instead.
+      // The issuer service is already tested implicitly through other tests.
+
+      // Test that issueOcapURL is called correctly directly
+      const kref = 'ko777';
+      vi.spyOn(mockRemoteComms, 'issueOcapURL').mockResolvedValue(
+        `ocap:issued@local-peer-id`,
+      );
+
+      const url = await ocapURLManager.issueOcapURL(kref);
+
+      expect(url).toBe('ocap:issued@local-peer-id');
+      expect(mockRemoteComms.issueOcapURL).toHaveBeenCalledWith(kref);
+    });
+
+    it('should throw error for non-remotable in issuer service', async () => {
+      const services = ocapURLManager.getServices();
+      const issuerService = services.issuerService.service as IssuerService;
+
+      // Plain object is not a remotable
+      const nonRemotable = { foo: 'bar' };
+
+      await expect(issuerService.issue(nonRemotable)).rejects.toThrow(
+        'Argument must be a remotable',
+      );
+    });
+
+    it('should handle undefined input in issuer service', async () => {
+      const services = ocapURLManager.getServices();
+      const issuerService = services.issuerService.service as IssuerService;
+
+      await expect(issuerService.issue(undefined)).rejects.toThrow(
+        'Argument must be a remotable',
+      );
+    });
+
+    it('should handle null input in issuer service', async () => {
+      const services = ocapURLManager.getServices();
+      const issuerService = services.issuerService.service as IssuerService;
+
+      await expect(issuerService.issue(null)).rejects.toThrow(
+        'Argument must be a remotable',
+      );
+    });
```

Comment:
Many of the tests deserve an `it.each`; whenever they are essentially input -> output tables, I find this improves readability.

```suggestion
    it.each([
      ['a non-remotable', { foo: 'bar'}],
      ['undefined', undefined],
      ['null', null],
    ])('throws on attempt to issue %s', (_desc, arg) => {
      const services = ocapURLManager.getServices();
      const issuerService = services.issuerService.service as IssuerService;

      await expect(issuerService.issue(arg)).rejects.toThrow(
        'Argument must be a remotable',
      );
    });